mava/types.py

```diff
@@ -21,17 +21,58 @@
 from optax._src.base import OptState
 from typing_extensions import NamedTuple, TypeAlias
 
-from mava.wrappers.jumanji import LogEnvState
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from flax.struct import dataclass
+
 
 Action: TypeAlias = chex.Array
 Value: TypeAlias = chex.Array
 Done: TypeAlias = chex.Array
 HiddenState: TypeAlias = chex.Array
-
-# Can't know the exact type of State or Timestep.
+# Can't know the exact type of State.
 State: TypeAlias = Any
-Observation: TypeAlias = Any
+
+
+class Observation(NamedTuple):
+    agents_view: chex.Array
+    action_mask: chex.Array
+    step_count: chex.Numeric
+
+
+class ObservationGlobalState(NamedTuple):
+    """The observation that the agent sees.
+    agents_view: the agents' view of other agents and shelves within their
+        sensor range. The number of features in the observation array
+        depends on the sensor range of the agent.
+    action_mask: boolean array specifying, for each agent, which action
+        (up, right, down, left) is legal.
+    global_state: the global state of the environment, which is the
+        concatenation of the agents' views.
+    step_count: the number of steps elapsed since the beginning of the episode.
+    """
```

Comment:
Note you may have to run pre-commit after adding this as I'm not sure about line lenghts

```suggestion
    """The observation that the agents see. Used in centralized systems.
    
    Same as `Observation` except it adds a `global_state` attribute for centralized training.
  
    global_state: the global state of the environment. (Often is concatenation of the agents' views.)
    """
```